app/backend/wells/views_v2.py

```diff
@@ -573,3 +577,170 @@ class WellDetail(WellDetailV1):
     This view is open to all, and has no permissions.
     """
     serializer_class = WellDetailSerializer
+
+
+class MislocatedWellsListView(ListAPIView):
+    """
+    API view to retrieve mislocated wells.
+    """
+    serializer_class = MislocatedWellsSerializer
+
+    swagger_schema = None
+    permission_classes = (WellsEditOrReadOnly,)
+    model = Well
+    pagination_class = APILimitOffsetPagination
+
+    # Allow searching on name fields, names of related companies, etc.
+    filter_backends = (WellListFilterBackend, BoundingBoxFilterBackend,
+                       filters.SearchFilter, WellListOrderingFilter, GeometryFilterBackend)
+    ordering = ('well_tag_number',)
+
+    def get_queryset(self):
+        """
+        This view should return a list of all mislocated wells
+        for the currently authenticated user.
+        """
+        queryset = Well.objects.all()
+
+        queryset = Well.objects.select_related('well_status').annotate(
```

Comment:
For clarification, here we are getting all the Well and related Well Status objects and updating the `work_start_date` and `work_end_date` fields. 

These fields already exist on the Well model, but we are updating them here because legacy wells may not have data for these fields?

Reply:
The Well model doesn't have these two fields available on it so we have to infer the correct start and end dates from the well_status object. The annotation just adds these fields to the queryset object so there's nothing actually being persisted to the database.

Reply:
Ok yes I see that now, I jumped to the wrong class in the file when searching 😄.